model/Core/Classes/ElementCollection.md

```diff
@@ -26,4 +26,9 @@ An SpdxCollection is a collection of Elements, not necessarily with unifying con
   - minCount: 1
 - imports
   - type: ExternalMap
-
+- profileConformance
+  - type: ProfileIdentifierType
+  - minCount: 1
+- profileNamespace
+  - type: ProfileIdentifierType
```

Comment:
I prefer to use enumerations to restrict the range of anyURI that is used.  We already have the profile enumerations defined.

Reply:
Yes, I understand the advantages of a restricted set of values.

On the other hand:

- there has to be a correspondence between the value `http://spdx.org/rdf/v3/Core/ProfileIdentifierType/software` and the prefix `http://spdx.org/rdf/v3/Software`, for a reader, for example, to check that only classes/properties of declared namespaces are being used... I tried to avoid using the words "namespace" and "map" for this 😝 
- allowing arbitrary URIs automatically takes care of all the `Extension` discussions we had back in the day, allowing people to add their own classes and properties.

Reply:
This puts namespaces into practice.  An application's conformance claim would be known by the namespaces it references, both within `http://spdx.org/rdf/v3/` and from non-SPDX definitions.

And to touch the third rail of namespace maps :-), applications can define any prefix strings they want without affecting any other applications: c:Element, s:Package, l:Apache-2.0, sec:VEXstuff, etc.  That's the beauty of maps, they are locally scoped, whether for software applications or for documents.  Adopting prefix string conventions is outside the scope of the model; the model just defines the prefix IRIs.

Reply:
If we want to be literal about the use of the actual namespaces use within the collection and the namespace property, I would go back to my original position that we should not have this property at all.

It isn't that much effort to traverse the elements and properties and collect what namespaces are actually used.

Even if we have the property, in practice it is quite possible we have a collection with a namespaceProfile declared at the collection level which isn't used as well as a no namespaceProfile declared for profiles that are used.  Not to mention, it is an optional property.

As a tool developer, I would never use the namespaceProfile myself due to the possibility of introducing errors and inconsistencies.